File: algorithms/easy/58_length_of_last_word.js
/**
 * 58. Length of Last Word
 * 
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {
    // Remove trailing and leading spaces, then split by spaces
    const words = s.trim().split(' ');
    
    // Return the length of the last word
    return words[words.length - 1].length;
};

// Alternative implementation without using split
var lengthOfLastWordAlt = function(s) {
    s = s.trim();  // Remove trailing and leading spaces
    let length = 0;
    
    // Start from the end and count characters until a space is encountered
    for (let i = s.length - 1; i >= 0; i--) {
        if (s[i] === ' ') {
            break;
        }
        length++;
    }
    
    return length;
};

// Test cases
console.log(lengthOfLastWord("Hello World"));         // 5
console.log(lengthOfLastWord("   fly me   to   the moon  "));  // 4
console.log(lengthOfLastWord("luffy is still joyboy"));  // 6

File: algorithms/easy/58_length_of_last_word.py
# 58. Length of Last Word

def length_of_last_word(s):
    """Return the length of the last word in a string.
    
    Args:
        s: String containing words and spaces
    
    Returns:
        Length of the last word
    """
    # Remove trailing and leading spaces, then split by spaces
    words = s.strip().split()
    
    # If there are no words, return 0
    if not words:
        return 0
    
    # Return the length of the last word
    return len(words[-1])

# Alternative implementation without using split
def length_of_last_word_alt(s):
    s = s.strip()  # Remove trailing and leading spaces
    length = 0
    
    # Start from the end and count characters until a space is encountered
    for i in range(len(s) - 1, -1, -1):
        if s[i] == ' ':
            break
        length += 1
    
    return length

# Test cases
print(length_of_last_word("Hello World"))         # 5
print(length_of_last_word("   fly me   to   the moon  "))  # 4
print(length_of_last_word("luffy is still joyboy"))  # 6